Adding documentation about SBOM report and how to get access to it (#1541)
aiven/devportal: docs/tools/cli/project.rst

.. code:: shell

  avn project user-remove --project my-project alice@example.com

Request project SBOM
--------------------

SBOM reports are generated per project and can be downloaded as long as the necessary permissions are set for the project. You can get the SBOM report download link for a project using the following command:

> ``avn_project_generate_sbom``

Reviewer: ```suggestion
``avn_project_generate_sbom``
'''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''

```

Reviewer: Hey, I'm sorry, I didn't realize this format was used throughout the article. My bad! I've added the heading line back in.

Author: no worries, it is added now :+1:

> ``avn project generate-sbom``

Reviewer: ```suggestion
```

Reviewer: Just deleting the duplicate.

Author: in that case I think I will keep the other one with the spaces in between words :slightly_smiling_face:

::

  avn project user-remove --project my-project alice@example.com

Request project SBOM
--------------------

> SBOM reports are generated per project and can be downloaded as long as the necessary permissions are set for the user/account.

Author: actually this is something I realized has changed. This was the initial idea, but the actual implementation check for permission in the project, so I will use `project` instead

.. _avn_project_generate_sbom:

``avn project generate-sbom``

> '''''''''''''''''''''''''''''

Author: the linter got mad at me when I didn't have that, I will try without it again and check if the linter complains

Gets the SBOM report download link for this project, in the chosen output format (CSV or SPDX).

.. list-table::
  :header-rows: 1
  :align: left

  * - Parameter
    - Information
  * - ``--project``

>     - The project to fetch details for

Reviewer: ```suggestion
    - The project name
```

  * - ``--output``

>     - Output extension (CSV or SPDX), to use

Reviewer: ```suggestion
    - Output format (CSV or SPDX)
```

> **Example:** Get the SBOM report download link for the project "my-project" in ``csv``.

Reviewer: ```suggestion
**Example:** Get the SBOM report download link for the project ``my-project`` in ``csv`` format:
```

> Downloads the SBOM report for this project, the SBOM is saved in the target file supplied by the user.
> 
> .. list-table::
>   :header-rows: 1
>   :align: left
> 
>   * - Parameter
>     - Information
>   * - ``--project``
>     - The project to fetch details for
>   * - ``--sbom-filepath``
>     - File name, including path and extension (CSV or SPDX), to use
> 
> **Example:** Download the SBOM report for the project "my-project", and save it in a file in the current directory called ``sbom.csv``.
> 
> ::
> 
>   avn project generate-sbom --project my-project --sbom-filepath sbom.csv

Reviewer 2: ```suggestion
Gets the SBOM report download link for this project, in the chosen output format (CSV or SPDX).

.. list-table::
  :header-rows: 1
  :align: left

  * - Parameter
    - Information
  * - ``--project``
    - The project to fetch details for
  * - ``--output``
    - Output extension (CSV or SPDX), to use

**Example:** Get the SBOM report download link for the project "my-project" in ``csv``.

::

  avn project generate-sbom --project my-project --output csv
```

Reviewer 2: As we modified the `aiven-client` feature + the [temporal external documentation located here](https://aiven.slab.com/posts/sbom-customer-external-doc-rvdyhhsa), suggesting the same changes here too.
